### DISCUSSION TITLE
IronPython fails to import a type which used to reside in system.dll

### DISCUSSION THREAD

**Discussion by mjkkirschner** [IronLanguages/ironpython3#1776] [General]
I am running into a behavior change from .net 4.8 to .net6+

We support our users writing running python in IronPython by hosting a python engine.

Previously in net48 the following code ran successfully:

```
import clr
from System.Diagnostics import Process
```
It nows throws an import exception about the type not existing in the module.

In net48 - this type was present in `system.dll / mscorlib` - it's now been moved to `system.diagnostics.process.dll` in net6+.

I note this snippet from the IronPython docs website.

`
IronPython only knows about assemblies that have been loaded using one of clr.AddReference-methods. It is possible for other assemblies to already be loaded before IronPython is loaded, or for other assemblies to be loaded by other parts of the application by calling [System.Reflection.Assembly.Load](http://msdn.microsoft.com/en-us/library/system.reflection.assembly.load.aspx), but IronPython will not be aware of these.
`

If I do `clr.AddReference("System.Diagnostics.Process")` of course it works - though the binary does seem to loaded already in my case which is confusing. (checking in the debug module window)

But - ideally I would like to enable our users to make the transition to net6 without needing to touch countless python programs.

1. Where does the DLR/IronPython actually load and import system.dll, system.private.core, etc so that it is available to `import` without using `clr.AddReference`.
2. As a client of the IronPython embedded API - is there an appropriate API to use to make the engine aware of this new dll (and other new dlls which move parts of the .net base class libraries out of system.dll).\
3. Any suggestions for how to make use of the net std shims that dotnet uses to redirect old code to the new type locations? `TypeForwardedTo` attribute. https://github.com/dotnet/runtime/blob/main/src/libraries/shims/netstandard/src/netstandard.cs

**Comment by slozier** [comment]
Assemblies are loaded into the `ScriptDomainManager` of the `LanguageContext`. You can use the `ScriptEngine.Runtime.LoadAssembly` API to load up an assembly to the domain manager of the runtime. As for using shims you can import them as you would any other DLL. For example, you could load the `System.dll` shim before any of the embedded code is run like this:
```c#
using System.Reflection;
using IronPython.Hosting;

var engine = Python.CreateEngine();
engine.Runtime.LoadAssembly(Assembly.Load("System"));
engine.Execute("import clr\nfrom System.Diagnostics import Process");
```
